src/downloader.js

```diff
@@ -0,0 +1,51 @@
+import isPromise from './utils/isPromise';
+
+/**
+ * Module for file downloading. Handle case, when you have to use custom downloading method
+ */
+export default class Downloader {
+  /**
+   * @param {object} params - downloader module params
+   * @param {ImageConfig} params.config - image tool config
+   * @param {Function} params.onDownload - callback which is called, when file is downloaded
+   * @param {Function} params.onError - callback for downloading errors
+   */
+  constructor({ config, onDownload, onError }) {
+    this.config = config;
+    this.onDownload = onDownload;
+    this.onError = onError;
+  }
+
+  /**
+   * Try to download file data and fill it using stored data
+   *
+   * @param {string} fileData - may be some key for custom downloading or url
+   */
+  download(fileData) {
```

Comment:
I'm not sure what is resolved here? url?

Reply:
changed doc